packaging/os/pulp_repo.py

```diff
@@ -0,0 +1,556 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2016, Joe Adams <adams10301@gmail.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: pulp_repo
+author: "Joe Adams (@sysadmind)"
+short_description: Add or remove Pulp repos from a remote host.
+description:
+   - Add or remove Pulp repos from a remote host.
+version_added: "2.2"
+requirements: []
+options:
+  name:
+    description:
+      - Name of the repo to add or remove. This correlates to repo-id in Pulp.
+    required: true
+  state:
+    description:
+      - The repo state. A state of C(sync) will queue a sync of the repo.
+        This is asynchronous but not delayed like a scheduled sync. A state of
+        C(publish) will use the repository's distributor to publish the content.
+    required: false
+    default: present
+    choices: [ "present", "absent", "sync", "publish" ]
+  pulp_host:
+    description:
+      - URL of the pulp server to connect to.
+    default: http://127.0.0.1
+  repo_type:
+    description:
+      - Repo plugin type to use (i.e. C(rpm), C(docker)).
+    default: rpm
+  feed_url:
+    description:
+      - Upstream feed URL to receive updates from.
+    required: false
+    default: null
+  proxy_host:
+    description:
+      - Proxy url setting for the pulp repository importer. This is in the
+        format scheme://host.
+    required: false
+    default: null
+  proxy_port:
+    description:
+      - Proxy port setting for the pulp repository importer.
+    required: false
+    default: null
+  importer_ssl_ca_cert:
+    description:
+      - CA certificate string used to validate the feed source SSL certificate.
+    required: false
+    default: null
+  importer_ssl_client_cert:
+    description:
+      - Certificate used as the client certificate when synchronizing the
+        repository. This is used to communicate authentication information to
+        the feed source. The value to this option must be the full path to the
+        certificate. The specified file may be the certificate itself or a
+        single file containing both the certificate and private key.
+    required: false
+    default: null
+  importer_ssl_client_key:
+    description:
+      - Private key to the certificate specified in importer_ssl_client_cert,
```

Comment:
Actually I'm not sure here if it should be `I()` or `C()`. According to the [documentation](http://docs.ansible.com/ansible/developing_modules.html#module-checklist):

> U() for urls, C() for files and options, I() for params, M() for modules

But I'm not sure what's _option_ and what's _parameter_. In the [`yum_repository`](https://github.com/ansible/ansible-modules-extras/blob/devel/packaging/os/yum_repository.py) module, I'm using `I()` for `name`, `state`, ... and `C()` for the values.

I would say that _parameter_ is the name of something and _option_ is the list of values which I can select from. But English is not my mother tongue so I can be wrong.